tencent-cloud/mysql/create.go

```diff
@@ -33,25 +35,60 @@ func NewCredential() (*common.Credential, *profile.ClientProfile) {
 	return credential, cpf
 }
 
-func (tm TcloudMySQL) Create() {
+func (tm TcloudMySQL) Create() (bool, error) {
+
+	// Init client
 	credential, cpf := NewCredential()
-	client, _ := cdb.NewClient(credential, tm.Region, cpf)
+	client, err := cdb.NewClient(credential, tm.Region, cpf)
+	if err != nil {
+		return false, err
+	}
 
+	// Check if instance with the same name already exists
+	exists, err := checkInstanceExists(client, tm.InstanceName)
+	if err != nil || exists {
+		return exists, errors.Wrap(err, "failed to check instances")
+	}
+
+	// Create instance
 	request := cdb.NewCreateDBInstanceHourRequest()
 	request.GoodsNum = common.Int64Ptr(tm.Instance)
 	request.Memory = common.Int64Ptr(tm.Memory)
 	request.Volume = common.Int64Ptr(tm.Volume)
 	request.Zone = common.StringPtr(tm.Zone)
 	response, err := client.CreateDBInstanceHour(request)
+	if _, ok := err.(*tencentErrors.TencentCloudSDKError); ok {
+		return false, errors.Wrap(err, "API error")
+	}
+	if err != nil {
+		return false, err
+	}
 
-	if _, ok := err.(*errors.TencentCloudSDKError); ok {
-		fmt.Printf("An API error has returned: %s", err)
-		return
+	// Parse response
+	b, err := json.Marshal(response.Response)
+	if err != nil {
+		// Want to return nil right here, because the instance has already
+		// been successfully created
+		fmt.Println(err)
+		return false, nil
 	}
-	// unexpected errors
+	fmt.Printf("%v", string(b))
+	return false, nil
+}
+
+func checkInstanceExists(client *cdb.Client, name string) (bool, error) {
+	req := cdb.NewDescribeDBInstancesRequest()
+	resp, err := client.DescribeDBInstances(req)
```

Comment:
there is no way to query for an individual DB name? is that a Go SDK issue or TCloud API issue?

Reply:
nope, you can check the api docs above

Reply:
RIP